### PR TITLE
[TEVA-2070] convert searchable collection to govuk standard

### PR DESCRIPTION
**Pull request by alexbowen**
https://dfedigital.atlassian.net/browse/TEVA-2070

## Changes in this PR:

- Convert to the Govuk component standard
- refactor test to allow for different variants when testing shared examples. Also i feel the test reads better as a test now as tests the variant and the options separately as they weren't clear before

### CODE REVIEW

File: app/components/flash_component.rb [hunk 1]
```diff
@@ -1,25 +1,25 @@
 class FlashComponent < GovukComponent::Base
-  attr_reader :variant, :message
+  attr_reader :variant_name, :message
 
   VARIANTS = %w[notice success warning alert].freeze
 
-  def initialize(variant:, message:, classes: [], html_attributes: {})
+  def initialize(variant_name:, message:, classes: [], html_attributes: {})
```
Comment (david-mears-dfe) [inline app/components/flash_component.rb:6]: You'll have to update application.html.slim as well to use variant_name

File: app/components/searchable_collection_component/searchable_collection_component.html+checkbox.slim [hunk 2]
```diff
@@ -1,15 +1,16 @@
-.searchable-collection-component class="#{border_class} #{scrollable_class}"
-  - if searchable
-    .searchable-collection-component__search
-      input.govuk-input.searchable-collection-component__search-input.icon.icon--left.icon--search placeholder="Search"
-      button.searchable-collection-component__search-button type="submit" aria-label="search options"
+= tag.div(class: classes, **html_attributes) do
+  div class="#{border_class} #{scrollable_class}"
```
Comment (david-mears-dfe) [inline app/components/searchable_collection_component/searchable_collection_component.html+checkbox.slim:2]: does it work if you put border_class and scrollable_class into `default_classes`? or is that a misuse of 'default'?

File: app/components/searchable_collection_component/searchable_collection_component.scss [hunk 3]
```diff
@@ -23,7 +23,7 @@
     }
   }
 
-  .searchable-collection-component__search {
+  &__search {
```
Comment (david-mears-dfe) [inline app/components/searchable_collection_component/searchable_collection_component.scss:26]: Is there a difference between `__` and `--` in these different class names?

File: spec/components/searchable_collection_component_spec.rb [hunk 4]
```diff
@@ -21,34 +23,45 @@
...
     end
 
     let!(:inline_component) { render_inline(radio_collection) }
 
-    it "a formbuilder radio button collection is used" do
-      expect(form).to have_received(:govuk_collection_radio_buttons)
-    end
-
-    it "is not searchable when collection is smaller than threshold" do
+    it "collection is not searchable" do
```
Comment (david-mears-dfe) [inline spec/components/searchable_collection_component_spec.rb:49]:
```suggestion
    it "is not searchable" do
```

File: spec/components/searchable_collection_component_spec.rb [hunk 5]
```diff
@@ -21,34 +23,45 @@
...
       expect(radio_collection.searchable).to be_falsey
       expect(inline_component.css(".searchable-collection-component__search")).to be_blank
       expect(inline_component.css(".searchable-collection-component--border")).to be_blank
     end
 
-    it "does not have small buttons when collection is smaller than threshold" do
+    it "collection items are large size" do
```
Comment (david-mears-dfe) [inline spec/components/searchable_collection_component_spec.rb:55]:
```suggestion
    it "has large collection items" do
```

File: spec/components/searchable_collection_component_spec.rb [hunk 6]
```diff
@@ -21,34 +23,45 @@
...
       expect(radio_collection.small).to be_falsey
     end
 
-    it "container is not scrollable when collection is smaller than threshold" do
+    it "collection is not scrollable" do
```
Comment (david-mears-dfe) [inline spec/components/searchable_collection_component_spec.rb:59]:
```suggestion
    it "is not scrollable" do
```

File: spec/components/searchable_collection_component_spec.rb [hunk 7]
```diff
@@ -57,21 +70,17 @@
 
     let!(:inline_component) { render_inline(checkbox_collection) }
 
-    it "a formbuilder checkbox collection is used" do
-      expect(form).to have_received(:govuk_collection_check_boxes)
-    end
-
-    it "is searchable when collection is equal to threshold" do
+    it "collection is searchable" do
```
Comment (david-mears-dfe) [inline spec/components/searchable_collection_component_spec.rb:73]:
```suggestion
    it "is searchable" do
```

File: spec/components/searchable_collection_component_spec.rb [hunk 8]
```diff
@@ -57,21 +70,17 @@
...
       expect(checkbox_collection.searchable).to be_truthy
       expect(inline_component.css(".searchable-collection-component__search").count).to eq(1)
       expect(inline_component.css(".searchable-collection-component--border").count).to eq(1)
     end
 
-    it "has buttons when collection is equal to threshold" do
+    it "collection items are small size" do
```
Comment (david-mears-dfe) [inline spec/components/searchable_collection_component_spec.rb:79]:
```suggestion
    it "has small collection items" do
```

File: spec/components/searchable_collection_component_spec.rb [hunk 9]
```diff
@@ -57,21 +70,17 @@
...
       expect(checkbox_collection.small).to be_truthy
     end
 
-    it "container is scrollable when collection is equal to threshold" do
+    it "collection is scrollable" do
```
Comment (david-mears-dfe) [inline spec/components/searchable_collection_component_spec.rb:83]:
```suggestion
    it "is scrollable" do
```

**Review by david-mears-dfe — APPROVED** [review]

File: app/components/searchable_collection_component/searchable_collection_component.html+checkbox.slim (hunk 2 above)
Comment (alexbowen) [reply app/components/searchable_collection_component/searchable_collection_component.html+checkbox.slim:2]: connor suggested once that if they are dependent on conditions it is a misuse of default, which i am inclined to agree with. but its much of a muchness tbh

File: app/components/searchable_collection_component/searchable_collection_component.scss (hunk 3 above)
Comment (alexbowen) [reply app/components/searchable_collection_component/searchable_collection_component.scss:26]: __ is a block, -- should signify a modifier (state). we dont use this to letter by any means but im trying to use them correctly particuarly in components

Comment (alexbowen) [reply app/components/searchable_collection_component/searchable_collection_component.scss:26]: it is an actual CSS convention, not something ive randomly invented, i should point out!

File: app/components/searchable_collection_component/searchable_collection_component.html+checkbox.slim (hunk 2 above)
Comment (david-mears-dfe) [reply app/components/searchable_collection_component/searchable_collection_component.html+checkbox.slim:2]: I think old connor is right.

**Pull request merged**